detect_os

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+UNAME_PATTERNS=( "\.el[0-9]+" "Ubuntu" )
+UNAME_MATCHES=( "rhel" "ubuntu" ) #always use lowercase
```

Comment:
You need to worry about amazon linux also.

Reply:
Assuming we're wanting to support Amazon Linux 2023 (or Amazon Linux 2)

We looking to support Amazon Linux 1 as well?

Reply:
How do we want to deal with OS major versions?  Package differences happen at major version boundaries (usually) - RHEL 8 and 9 was a fun jump, and 10 promises to be the same.

We'll want to support Amazon Linux 2 and Amazon Linux 2023 - no reason to worry about Amazon Linux 1.  I'm not sure we've ever even spun up an instance with that.

Reply:
I can add a version flag for fetching major version numbers, seems like the easiest thing to do/deal with